utils/test_mocks.go

```diff
@@ -31,19 +35,68 @@ func CreateXscRestsMockServer(t *testing.T, testHandler restsTestHandler) (*http
 	return testServer, serverDetails, serviceManager
 }
 
+func CreateXrayRestsMockServer(testHandler restsTestHandler) (*httptest.Server, *config.ServerDetails) {
+	testServer := CreateRestsMockServer(testHandler)
+	serverDetails := &config.ServerDetails{Url: testServer.URL + "/", XrayUrl: testServer.URL + "/xray/"}
+	return testServer, serverDetails
+}
+
 func XscServer(t *testing.T, xscVersion string) (*httptest.Server, *config.ServerDetails) {
 	serverMock, serverDetails, _ := CreateXscRestsMockServer(t, func(w http.ResponseWriter, r *http.Request) {
 		if r.RequestURI == "/xsc/api/v1/system/version" {
 			_, err := w.Write([]byte(fmt.Sprintf(`{"xsc_version": "%s"}`, xscVersion)))
-			if err != nil {
+			if !assert.NoError(t, err) {
 				return
 			}
 		}
 		if r.RequestURI == "/xsc/api/v1/event" {
 			if r.Method == http.MethodPost {
 				w.WriteHeader(http.StatusCreated)
 				_, err := w.Write([]byte(fmt.Sprintf(`{"multi_scan_id": "%s"}`, TestMsi)))
-				if err != nil {
+				if !assert.NoError(t, err) {
+					return
+				}
+			}
+		}
+		if r.RequestURI == "/xsc/api/v1/profile/"+TestConfigProfileName {
+			if r.Method == http.MethodGet {
+				w.WriteHeader(http.StatusOK)
+				content, err := os.ReadFile("../../tests/testdata/other/configProfile/configProfileExample.json")
+				if !assert.NoError(t, err) {
+					return
+				}
+				_, err = w.Write(content)
+				if !assert.NoError(t, err) {
+					return
+				}
+			}
+		}
+	})
+	return serverMock, serverDetails
+}
+
+func XrayServer(t *testing.T, xrayVersion string) (*httptest.Server, *config.ServerDetails) {
+	serverMock, serverDetails := CreateXrayRestsMockServer(func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI == "/xray/api/v1/system/version" {
```

Comment:
```suggestion
		if r.RequestURI == fmt.Sprintf(versionApiUrl, "xray) "/xray/api/v1/system/version" {
```
since we have similar API in xray and xsc you can create consts and fmt.Sprintf the values
`const versionApiUrl = "/%s/api/v1/system/version"`